src-electron/upgrade/upgrade.js

```diff
@@ -40,14 +48,48 @@ async function getForcedExternalStorage(db, attributeId) {
   let pkgs = await queryPackage.getPackageRefByAttributeId(db, attributeId)
   let zcl = await queryPackage.getPackageByPackageId(db, pkgs)
   zcl = zcl.path
-  let obj = await fsp.readFile(zcl)
-  let data = JSON.parse(obj)
+  let obj = await fsp.readFile(zcl, 'utf-8')
+  let data
+  if (isJsonString(obj)) {
+    data = JSON.parse(obj)
+  }
   let byName = data?.attributeAccessInterfaceAttributes
   let lists = data?.listsUseAttributeAccessInterface
   let forcedExternal = { byName, lists }
   return forcedExternal
 }
 
+/**
+ * Returns a list of attributes with proper global attributes
+ *
+ * @export
+ * @param {*} db
+ * @param {*} clusterRef
+ * @param {*} attributes
```

Comment:
This is ... not helpful documentation either.  `attributes` is expected to be an array of objects representing attributes we queried from the db, right?  clusterRef should be the database id (not spec id) of the cluster, I assume?